pkg/executor/show.go

```diff
@@ -805,6 +807,14 @@ func (e *ShowExec) fetchShowIndex() error {
 		if colStats != nil {
 			ndv = colStats.NDV
 		}
+
+		pi := tb.Meta().GetPartitionInfo()
+		isPartitioned := pi != nil && pi.Type != pmodel.PartitionTypeNone
+
+		if isPartitioned && e.Ctx().GetSessionVars().EnableGlobalIndex {
+			isGlobalIndex = "YES"
```

Comment:
If PKIsHandle, then it is by definition a LOCAL index, i.e. each partition will have the clustered index, and the PK column must be a part of the partitioning expression.

Test by:
```sql
set tidb_enable_global_index = 0;
create table t (a int primary key clustered, b int) partition by hash (a) partitions 2;
show create table t;
show index from t;
set tidb_enable_global_index = 1;
show index from t;
```